flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriterTest.java

```diff
@@ -214,6 +282,73 @@ public void testWriteData() throws Exception {
         assertThat(writer.rowDataCollectors).isEmpty();
     }
 
+    @Test
+    public void testWriteDataWithNullTimestamp() throws Exception {
+        final MockedSinkWriter writer = new MockedSinkWriter();
+        final ReducingUpsertWriter<?> bufferedWriter = createBufferedWriter(writer);
+
+        // write 4 records which doesn't trigger batch size
```

Comment:
Why doesn't this trigger batching? Are we relying on some default config value here?